external/model-preparation-algorithm/tests/test_xai.py

```diff
@@ -2,146 +2,236 @@
-            backbone=dict(
-                type="mobilenetv2_w1", out_indices=(4, 5), frozen_stages=-1, norm_eval=False, pretrained=True
-            ),
-            bbox_head=dict(
-                type="CustomSSDHead",
-                in_channels=(96, 320),
-                num_classes=20,
-                anchor_generator=dict(
-                    type="SSDAnchorGeneratorClustered",
-                    reclustering_anchors=True,
-                    strides=[16, 32],
-                    widths=[
-                        np.array([70.93408016, 132.06659281, 189.56180207, 349.90057837]),
-                        np.array([272.31733885, 448.52200666, 740.63350023, 530.78990182, 790.99297377]),
-                    ],
-                    heights=[
-                        np.array([93.83759764, 235.21261441, 432.6029996, 250.08979657]),
-                        np.array([672.8829653, 474.84783528, 420.18291446, 741.02592293, 766.45636125]),
-                    ],
-                ),
-                bbox_coder=dict(
-                    type="DeltaXYWHBBoxCoder", target_means=[0.0, 0.0, 0.0, 0.0], target_stds=[0.1, 0.1, 0.2, 0.2]
-                ),
-            ),
-        )
-        model = build_detector(model_cfg)
-        return model.eval()
 
-    def test_recipro_cam(self):
-        model = self.get_classification_model()
+def saliency_maps_check(predicted_dataset, task_labels, assert_text, only_predicted=False):
+    for data_point in predicted_dataset:
+        saliency_map_counter = 0
+        metadata_list = data_point.get_metadata()
+        for metadata in metadata_list:
+            if isinstance(metadata.data, ResultMediaEntity):
+                if metadata.data.type == "saliency_map":
+                    saliency_map_counter += 1
+                    assert metadata.data.numpy.ndim == 3
+                    assert metadata.data.numpy.shape == (data_point.height, data_point.width, 3)
+        if only_predicted:
+            assert saliency_map_counter == len(data_point.get_roi_labels(task_labels)), assert_text
+        else:
+            assert saliency_map_counter == len(task_labels), assert_text
+
+
+class TestExplainMethods:
+    ref_saliency_vals_cls = {
+        "EfficientNet-B0": np.array([36, 185, 190, 159, 173, 124, 19], dtype=np.uint8),
+        "MobileNet-V3-large-1x": np.array([21, 38, 56, 134, 100, 41, 38], dtype=np.uint8),
+        "EfficientNet-V2-S": np.array([166, 204, 201, 206, 218, 221, 138], dtype=np.uint8),
+    }
+
+    ref_saliency_shapes = {
+        "ATSS": (2, 4, 4),
+        "SSD": (81, 13, 13),
+        "YOLOX": (80, 13, 13),
+    }
+
+    ref_saliency_vals_det = {
+        "ATSS": np.array([78, 217, 42, 102], dtype=np.uint8),
+        "SSD": np.array([225, 158, 221, 106, 146, 158, 227, 149, 137, 135, 200, 159, 255], dtype=np.uint8),
+        "YOLOX": np.array([109, 174, 82, 214, 178, 184, 168, 161, 163, 156, 220, 233, 195], dtype=np.uint8),
+    }
+
+    @e2e_pytest_component

```

Comment:
we can use `@e2e_pytest_api` for API tests

Reply:
This one in particular is not an API test. I have added `@e2e_pytest_api` decorators to all API tests. Please take a look

Reply:
I'm not sure, but the issue with tests not being collected could be a problem with the develop branch, not just this PR. However, when I looked at the test builds last week, I noticed that they collected just fine. I will share this with the group and share any progress.

Reply:
@harimkang 
Hi Harim, do you have any update on this?

Here is the latest build https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/849/
I can confirm that from MPA only `ote_cli/` tests running, the rest are not running (including XAI test). Not sure I know the reason.